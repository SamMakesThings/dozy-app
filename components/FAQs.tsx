import React, { useCallback } from 'react';
import {
  View,
  Text,
  StyleSheet,
  FlatList,
  ViewProps,
  TouchableOpacity,
  TouchableOpacityProps,
  ListRenderItemInfo,
} from 'react-native';
import { withTheme } from '@draftbit/ui';
import { scale } from 'react-native-size-matters';
import { Theme } from '../types/theme';
import { FAQContent } from '../types/faq';

interface QuestionProps extends TouchableOpacityProps {
  content: string;
  theme: Theme;
}

const Question: React.FC<QuestionProps> = ({
  content,
  theme,
  style,
  ...props
}) => (
  <TouchableOpacity style={[styles.questionContainer, style]} {...props}>
    <Text style={[theme.typography.body1, { color: theme.colors.surface }]}>
      {content}
    </Text>
  </TouchableOpacity>
);

export interface FAQsProps extends ViewProps {
  theme: Theme;
  title: string;
  questions: { id: string; question: FAQContent }[];
  onQuestionClick: (questionId: string) => void;
}

export const FAQs: React.FC<FAQsProps> = ({
  theme,
  title,
  questions,
  onQuestionClick,
  style,
  ...props
}) => {
  const renderItem = useCallback(
    ({ item }: ListRenderItemInfo<{ id: string; question: FAQContent }>) => (
      <Question
        content={item.question.content}
        theme={theme}
        onPress={() => onQuestionClick(item.id)}
      />
    ),
    [theme, onQuestionClick],
  );

  return (
    <View style={[styles.container, style]} {...props}>
      <Text
        style={[
          styles.title,
          theme.typography.headline5,
          {
            color: theme.colors.secondary,
          },
        ]}
      >
        {title}
      </Text>
      <FlatList
        style={styles.list}
        data={questions}
        renderItem={renderItem}
        keyExtractor={(item) => item.id}
        ItemSeparatorComponent={() => <View style={styles.separator} />}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignSelf: 'stretch',
    marginHorizontal: -6,
  },
  title: {
    padding: 6,
  },
  list: {
    flex: 1,
    marginTop: scale(20),
  },
  // eslint-disable-next-line react-native/no-color-literals
  questionContainer: {
    paddingLeft: scale(36),
    paddingTop: scale(12),
    paddingRight: scale(21),
    paddingBottom: scale(10),
    backgroundColor: '#404b69',
    borderRadius: 3,
  },
  separator: {
    height: scale(12),
  },
});

export default withTheme(FAQs);
