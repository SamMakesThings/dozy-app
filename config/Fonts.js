export default {
  RubikRegular: require("../assets/fonts/RubikRegular.ttf"),
  RubikMedium: require("../assets/fonts/RubikMedium.ttf"),
  RubikBold: require("../assets/fonts/RubikBold.ttf")
}
