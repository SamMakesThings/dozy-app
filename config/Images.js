export default {
  Model016: require('../assets/images/model-016.png'),
  UndrawBed: require('../assets/images/UndrawBed.png'),
  WaveHello: require('../assets/images/WaveHello.png'),
  Slumberwordmark: require('../assets/images/Slumberwordmark.png'),
  DreamBg: require('../assets/images/DreamBg.png'),
  WordmarkTrans: require('../assets/images/WordmarkTrans.png'),
  DreamBgExtended: require('../assets/images/DreamBgExtended.png'),
  WomanInBed: require('../assets/images/WomanInBed.jpeg')
};
