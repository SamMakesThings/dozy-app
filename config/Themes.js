import { systemWeights } from 'react-native-typography';

export const slumber_theme = {
  disabledOpacity: 0.5,
  spacing: {
    gutters: 16,
    text: 4,
    small: 8,
    medium: 12,
    large: 16
  },
  borderRadius: {
    global: 6,
    button: 36
  },
  typography: {
    body1: {
      fontSize: 16,
      lineHeight: 26,
      fontFamily: 'RubikRegular'
    },
    body2: {
      fontSize: 14,
      lineHeight: 22,
      fontFamily: 'RubikRegular'
    },
    button: {
      fontSize: 16,
      lineHeight: 16,
      fontFamily: 'RubikRegular'
    },
    caption: {
      fontSize: 12,
      lineHeight: 16,
      fontFamily: 'RubikRegular'
    },
    headline1: {
      fontSize: 60,
      lineHeight: 71,
      fontFamily: 'RubikBold'
    },
    headline2: {
      fontSize: 48,
      lineHeight: 58,
      fontFamily: 'RubikMedium'
    },
    headline3: {
      fontSize: 38,
      lineHeight: 55,
      fontFamily: 'RubikMedium'
    },
    headline4: {
      fontSize: 24,
      lineHeight: 32,
      fontFamily: 'RubikBold'
    },
    headline5: {
      fontSize: 20,
      lineHeight: 26,
      fontFamily: 'RubikBold'
    },
    headline6: {
      fontSize: 16,
      lineHeight: 24,
      fontFamily: 'RubikBold'
    },
    overline: {
      fontSize: 12,
      lineHeight: 16,
      letterSpacing: 2,
      fontFamily: 'RubikRegular'
    },
    subtitle1: {
      fontSize: 16,
      lineHeight: 26,
      fontFamily: 'RubikRegular'
    },
    subtitle2: {
      fontSize: 14,
      lineHeight: 22,
      fontFamily: 'RubikRegular'
    }
  },
  colors: {
    background: 'rgba(35, 43, 63, 1)',
    divider: 'rgba(234, 237, 242, 1)',
    error: 'rgba(255, 69, 100, 1)',
    light: 'rgba(153, 172, 185, 1)',
    lightInverse: 'rgba(255, 255, 255, 0.68)',
    medium: 'rgba(64, 75, 105, 1)',
    mediumInverse: 'rgba(255, 255, 255, 0.87)',
    primary: 'rgba(0, 129, 138, 1)',
    secondary: 'rgba(219, 237, 243, 1)',
    strong: 'rgba(255, 255, 255, 1)',
    strongInverse: 'rgba(255, 255, 255, 1)',
    surface: 'rgba(255, 255, 255, 1)'
  },
  elevation: {
    '0': {
      shadowColor: 'rgba(255, 255, 255, 1)',
      shadowOffset: {
        width: 0,
        height: 0
      },
      shadowRadius: 0,
      shadowOpacity: 0,
      borderWidth: 0,
      borderColor: 'rgba(255, 255, 255, 1)',
      borderOpacity: 0
    },
    '1': {
      shadowColor: 'rgba(255, 255, 255, 1)',
      shadowOffset: {
        width: 0,
        height: 2
      },
      shadowRadius: 4,
      shadowOpacity: 0.06,
      borderWidth: 1,
      borderColor: 'rgba(255, 255, 255, 1)',
      borderOpacity: 0.06
    },
    '2': {
      shadowColor: 'rgba(255, 255, 255, 1)',
      shadowOffset: {
        width: 0,
        height: 2
      },
      shadowRadius: 4,
      shadowOpacity: 0.2,
      borderWidth: 0,
      borderColor: 'rgba(255, 255, 255, 1)',
      borderOpacity: 0
    },
    '3': {
      shadowColor: 'rgba(255, 255, 255, 1)',
      shadowOffset: {
        width: 0,
        height: 6
      },
      shadowRadius: 6,
      shadowOpacity: 0.12,
      borderWidth: 0,
      borderColor: 'rgba(255, 255, 255, 1)',
      borderOpacity: 0
    }
  }
};
