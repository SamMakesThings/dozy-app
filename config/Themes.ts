import { scale } from 'react-native-size-matters';
import { Theme } from '../types/theme';

// eslint-disable-next-line @typescript-eslint/naming-convention
export const dozy_theme: Theme = {
  disabledOpacity: 0.5,
  spacing: {
    gutters: 16,
    text: 4,
    small: 8,
    medium: 12,
    large: 16,
  },
  borderRadius: {
    global: 6,
    button: 36,
  },
  buttonLayout: {
    minHeight: scale(47),
    justifyContent: 'center',
  },
  typography: {
    smallLabel: {
      fontSize: scale(16),
      lineHeight: scale(26),
      fontFamily: 'RubikRegular',
    },
    body1: {
      fontSize: scale(18),
      lineHeight: scale(22),
      fontFamily: 'RubikRegular',
      letterSpacing: -0.25,
    },
    body2: {
      fontSize: scale(14),
      lineHeight: scale(22),
      fontFamily: 'RubikRegular',
    },
    button: {
      fontSize: scale(16),
      lineHeight: scale(16),
      fontFamily: 'RubikRegular',
    },
    caption: {
      fontSize: scale(12),
      lineHeight: scale(16),
      fontFamily: 'RubikRegular',
    },
    headline1: {
      fontSize: scale(60),
      lineHeight: scale(71),
      fontFamily: 'RubikBold',
    },
    headline2: {
      fontSize: scale(48),
      lineHeight: scale(58),
      fontFamily: 'RubikMedium',
    },
    headline3: {
      fontSize: scale(28),
      lineHeight: scale(35),
      fontFamily: 'RubikMedium',
    },
    headline4: {
      fontSize: scale(24),
      lineHeight: scale(32),
      fontFamily: 'RubikBold',
    },
    headline5: {
      fontSize: scale(22),
      lineHeight: scale(26),
      fontFamily: 'RubikMedium',
    },
    headline6: {
      fontSize: scale(16),
      lineHeight: scale(24),
      fontFamily: 'RubikMedium',
    },
    cardTitle: {
      fontSize: scale(20),
      lineHeight: scale(26),
      fontFamily: 'RubikMedium',
    },
    overline: {
      fontSize: scale(12),
      lineHeight: scale(16),
      letterSpacing: 2,
      fontFamily: 'RubikRegular',
    },
    subtitle1: {
      fontSize: scale(16),
      lineHeight: scale(26),
      fontFamily: 'RubikRegular',
    },
    subtitle2: {
      fontSize: scale(14),
      lineHeight: scale(22),
      fontFamily: 'RubikRegular',
    },
  },
  colors: {
    background: 'rgba(35, 43, 63, 1)',
    divider: 'rgba(234, 237, 242, 1)',
    error: 'rgba(255, 69, 100, 1)',
    light: 'rgba(153, 172, 185, 1)',
    lightInverse: 'rgba(153, 172, 185, 0.3)',
    medium: 'rgba(64, 75, 105, 1)',
    mediumInverse: 'rgba(255, 255, 255, 0.87)',
    primary: 'rgba(0, 129, 138, 1)',
    secondary: 'rgba(219, 237, 243, 1)',
    strong: 'rgba(255, 255, 255, 1)',
    strongInverse: 'rgba(255, 255, 255, 1)',
    surface: 'rgba(255, 255, 255, 1)',
  },
  elevation: {
    0: {
      shadowColor: 'rgba(18, 20, 44, 1)',
      shadowOffset: {
        width: 0,
        height: 0,
      },
      shadowRadius: 0,
      shadowOpacity: 0,
      borderWidth: 0,
      borderColor: 'rgba(18, 20, 44, 1)',
      borderOpacity: 0,
    },
    1: {
      shadowColor: 'rgba(18, 20, 44, 1)',
      shadowOffset: {
        width: 0,
        height: 2,
      },
      shadowRadius: 4,
      shadowOpacity: 0.06,
      borderWidth: 1,
      borderColor: 'rgba(18, 20, 44, 1)',
      borderOpacity: 0.06,
    },
    2: {
      shadowColor: 'rgba(18, 20, 44, 1)',
      shadowOffset: {
        width: 0,
        height: 2,
      },
      shadowRadius: 4,
      shadowOpacity: 0.2,
      borderWidth: 0,
      borderColor: 'rgba(18, 20, 44, 1)',
      borderOpacity: 0,
    },
    3: {
      shadowColor: 'rgba(18, 20, 44, 1)',
      shadowOffset: {
        width: 0,
        height: 6,
      },
      shadowRadius: 6,
      shadowOpacity: 0.12,
      borderWidth: 0,
      borderColor: 'rgba(18, 20, 44, 1)',
      borderOpacity: 0,
    },
  },
};
