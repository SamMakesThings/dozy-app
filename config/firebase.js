import * as firebase from 'firebase';
  
firebase.initializeApp(Expo.Constants.manifest.extra.firebase);

export default firebase;