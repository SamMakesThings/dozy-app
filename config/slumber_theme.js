import { systemWeights } from 'react-native-typography'

export const slumber_theme = {
  "disabledOpacity": 0.5,
  "spacing": {
    "gutters": 16,
    "text": 4,
    "small": 8,
    "medium": 12,
    "large": 16
  },
  "borderRadius": {
    "global": 3,
    "button": 24
  },
  "typography": {
    "body1": {
      "fontSize": 16,
      "lineHeight": 26,
      "letterSpacing": 0,
      "fontWeight": null,
      "fontFamily": "System"
    },
    "body2": {
      "fontSize": 14,
      "lineHeight": 22,
      "letterSpacing": 0,
      "fontWeight": null,
      "fontFamily": "System"
    },
    "button": {
      "fontSize": 14,
      "lineHeight": 16,
      "letterSpacing": 0,
      "fontWeight": null,
      "fontFamily": "System"
    },
    "caption": {
      "fontSize": 12,
      "lineHeight": 16,
      "letterSpacing": 0,
      "fontWeight": null,
      "fontFamily": "System"
    },
    "headline1": {
      "fontSize": 60,
      "lineHeight": 71,
      "letterSpacing": 0,
      "fontWeight": null,
      "fontFamily": "System"
    },
    "headline2": {
      "fontSize": 48,
      "lineHeight": 58,
      "letterSpacing": 0,
      "fontWeight": null,
      "fontFamily": "System"
    },
    "headline3": {
      "fontSize": 34,
      "lineHeight": 40,
      "letterSpacing": 0,
      "fontWeight": null,
      "fontFamily": "System"
    },
    "headline4": {
      "fontSize": 24,
      "lineHeight": 32,
      "letterSpacing": 0,
      ...systemWeights.semibold
    },
    "headline5": {
      "fontSize": 20,
      "lineHeight": 26,
      "letterSpacing": 0,
      "fontWeight": null,
      "fontFamily": "System"
    },
    "headline6": {
      "fontSize": 16,
      "lineHeight": 24,
      "letterSpacing": 0,
      ...systemWeights.semibold
    },
    "overline": {
      "fontSize": 12,
      "lineHeight": 16,
      "letterSpacing": 2,
      "fontWeight": null,
      "fontFamily": "System"
    },
    "subtitle1": {
      "fontSize": 16,
      "lineHeight": 26,
      "letterSpacing": 0,
      "fontWeight": null,
      "fontFamily": "System"
    },
    "subtitle2": {
      "fontSize": 14,
      "lineHeight": 22,
      "letterSpacing": 0,
      "fontWeight": null,
      "fontFamily": "System"
    }
  },
  "colors": {
    "background": "rgba(35, 43, 63, 1)",
    "divider": "rgba(234, 237, 242, 1)",
    "error": "rgba(255, 69, 100, 1)",
    "light": "rgba(153, 172, 185, 1)",
    "lightInverse": "rgba(255, 255, 255, 0.68)",
    "medium": "rgba(64, 75, 105, 1)",
    "mediumInverse": "rgba(255, 255, 255, 0.87)",
    "primary": "rgba(0, 129, 138, 1)",
    "secondary": "rgba(219, 237, 243, 1)",
    "strong": "rgba(255, 255, 255, 1)",
    "strongInverse": "rgba(255, 255, 255, 1)",
    "surface": "rgba(255, 255, 255, 1)"
  },
  "elevation": {
    "0": {
      "shadowColor": "rgba(18, 20, 44, 1)",
      "shadowOffset": {
        "width": 0,
        "height": 0
      },
      "shadowRadius": 0,
      "shadowOpacity": 0,
      "borderWidth": 0,
      "borderColor": "rgba(18, 20, 44, 1)",
      "borderOpacity": 0
    },
    "1": {
      "shadowColor": "rgba(18, 20, 44, 1)",
      "shadowOffset": {
        "width": 0,
        "height": 2
      },
      "shadowRadius": 4,
      "shadowOpacity": 0.06,
      "borderWidth": 1,
      "borderColor": "rgba(18, 20, 44, 1)",
      "borderOpacity": 0.06
    },
    "2": {
      "shadowColor": "rgba(18, 20, 44, 1)",
      "shadowOffset": {
        "width": 0,
        "height": 2
      },
      "shadowRadius": 4,
      "shadowOpacity": 0.2,
      "borderWidth": 0,
      "borderColor": "rgba(18, 20, 44, 1)",
      "borderOpacity": 0
    },
    "3": {
      "shadowColor": "rgba(18, 20, 44, 1)",
      "shadowOffset": {
        "width": 0,
        "height": 6
      },
      "shadowRadius": 6,
      "shadowOpacity": 0.12,
      "borderWidth": 0,
      "borderColor": "rgba(18, 20, 44, 1)",
      "borderOpacity": 0
    }
  }
}