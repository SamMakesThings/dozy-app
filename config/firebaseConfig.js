import * as firebase from 'firebase';
import Expo from 'expo';
import Constants from 'expo-constants';
  
export const FbApp = !firebase.apps.length ? firebase.initializeApp(Constants.manifest.extra.firebase) : firebase.app();

export const FbAuth = firebase.auth();

export const FbLib = firebase;

// Making Firestore happen, testing data retrieval
/* 
var db = firebase.firestore();

var docRef = db.collection("cities").doc("SF");

docRef.get().then(function(doc) {
    if (doc.exists) {
        console.log("Document data:", doc.data());
    } else {
        // doc.data() will be undefined in this case
        console.log("No such document!");
    }
}).catch(function(error) {
    console.log("Error getting document:", error);
});
*/

// Google sign-in functionality
// const clientId = '713165282203-jjc54if1n7krahda9gvkio0siqltq57t.apps.googleusercontent.com';