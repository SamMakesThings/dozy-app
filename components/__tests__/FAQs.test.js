import React from 'react';
import { shallow } from 'enzyme';
import { FAQs } from '../FAQs';
import { dozy_theme } from '../../config/Themes';

describe('FAQs', () => {
  it('should match the snapshot', () => {
    const component = shallow(
      <FAQs
        theme={dozy_theme}
        title="What would you like to know?"
        questions={[
          { question: 'question 1', answer: 'answer 1' },
          { question: 'question 2', answer: 'answer 2' },
          { question: 'question 3', answer: 'answer 3' },
          { question: 'question 4', answer: 'answer 4' },
        ]}
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
