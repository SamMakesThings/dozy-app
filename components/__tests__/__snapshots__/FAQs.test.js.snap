// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQs should match the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "marginHorizontal": -6,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "padding": 6,
        },
        Object {
          "fontFamily": "RubikMedium",
          "fontSize": 47.14285714285714,
          "lineHeight": 55.714285714285715,
        },
        Object {
          "color": "rgba(219, 237, 243, 1)",
        },
      ]
    }
  >
    What would you like to know?
  </Text>
  <FlatList
    ItemSeparatorComponent={[Function]}
    data={
      Array [
        Object {
          "answer": "answer 1",
          "question": "question 1",
        },
        Object {
          "answer": "answer 2",
          "question": "question 2",
        },
        Object {
          "answer": "answer 3",
          "question": "question 3",
        },
        Object {
          "answer": "answer 4",
          "question": "question 4",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    style={
      Object {
        "flex": 1,
        "marginTop": 42.857142857142854,
      }
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
